bot/university/university.py

```diff
@@ -0,0 +1,378 @@
+        reached, the embed will be removed and the bot will post a message to inform the user about what happened.
+
+        Args:
+            author (discord.User): The user who invoked the command.
+            channel (discord.TextChannel): The channel in which the command has been invoked.
+            persons (List[ET.Element]): A list containing the person elements from the XML string returned by `ufind`.
+
+        Returns:
+            int: The index of the selected person.
+        """
+        timeout = 15.0
+        list_selection_emojis = SelectionEmoji.to_list()
+
+        embed = _create_embed_staff_selection(persons)
+        message = await channel.send(embed=embed)
+
+        for i in range(len(embed.fields)):
+            await message.add_reaction(list_selection_emojis[i])
+
+        def check(_reaction, user):
+            return user == author and SelectionEmoji(_reaction.emoji) is not None
+
+        await message.delete(delay=timeout)
+        reaction = await self.bot.wait_for('reaction_add', timeout=timeout + 0.1, check=check)
+        await message.delete()
+
+        return list_selection_emojis.index(reaction[0].emoji)
+
+
+def _create_embed_staff_selection(persons: List[ET.Element]) -> discord.Embed:
+    """Method which creates the embed for selecting a staff member.
+
+    Args:
+        persons (List[ET.Element]): A list containing the individual person elements from the XML string received by
+        `ufind`.
+
+    Returns:
+        discord.Embed: Embedded message which lists all of the staff members returned by `ufind`.
+    """
+    list_selection_emojis = SelectionEmoji.to_list()
+    description = "Sollte die gesuchte Person hier nicht aufgelistet sein, dann versuche bitte deine Suchanfrage zu " \
+                  "spezifizieren.\n" + constants.ZERO_WIDTH_SPACE
+
+    embed = discord.Embed(color=constants.EMBED_COLOR_SELECTION, title="Wähle einen Mitarbeiter aus:",
+                          description=description)
+
+    for index, person in enumerate(persons):
+        name = "{0} {1} {2}".format(list_selection_emojis[index], person.findtext("firstname"),
+                                    person.findtext("lastname"))
+        embed.add_field(name=name, inline=True, value=constants.ZERO_WIDTH_SPACE)
+
+    return embed
+
+
+def _parse_staff_xml(str_xml: str) -> Dict[str, Union[datetime, str, None]]:
+    """Method for parsing the staff member data returned by `ufind` in XML.
+
+    Parses and prepares the data about a staff member and returns it in a dictionary so it can be posted in an
+    embed later.
+
+    Args:
+        str_xml (str): String containing data about a staff member in XML.
+
+    Returns:
+        Dict[str, str]: A dictionary containing the wanted information about a staff member.
+    """
+    xml = ET.fromstring(str_xml)
+
+    str_title = _create_staff_embed_title(xml.find("title"), xml.findtext("firstname"), xml.findtext("lastname"))
```

Comment:
Why do all methods contain the word embed?

Reply:
Because the strings will be used for the staff embeds (the embedded messages which contain information about a staff member). So basically this is the method for creating the title for the staff embed.

Reply:
If you have a better naming scheme then let me know.

Reply:
But the code inside is completely separated from the embed. I could use the string for anything else that's not an embed. I'd just throw 'embed' out of the name, but that's not high priority

Reply:
Yeah you could, but the string is created/formatted especially for the staff embed. I doubt that we will need this string in exactly this way for something other than this.